src/explore-education-statistics-common/src/modules/charts/components/utils/__tests__/generateLegendDataClasses.test.ts

```diff
@@ -0,0 +1,1145 @@
+import generateLegendDataClasses, {
+  LegendDataClass,
+} from '@common/modules/charts/components/utils/generateLegendDataClasses';
+
+describe('generateLegendDataClasses', () => {
```

Comment:
This needs tests for when the values have more than 2 decimal places and if decimalPlaces is passed in.

Reply:
Good catch, added!